test/internal/modules/billing/services/transactions-csv.js

```diff
@@ -7,166 +7,250 @@ const transactionsCSV = require('internal/modules/billing/services/transactions-
-      licenceNumber: '01/234/567'
-    }
-  }],
-  invoiceAccount: {
-    id: 'a764b68a-0167-4f8c-82b1-da08824c2683',
-    accountNumber: 'A12345678A',
-    company: {
-      id: '530d40da-7355-44ba-9468-5500d3e8fe47',
-      type: 'organisation',
-      name: 'South West Water Ltd'
+      address: {
+        id: '6f9dfbd6-b534-442c-bf30-cb32a53b9a6c',
+        town: 'Apple',
+        county: 'Appleshire',
+        country: null,
+        postcode: 'AP9 8RG',
+        addressLine1: 'Little  Orchard',
+        addressLine2: 'Orchard lane',
+        addressLine3: 'Orchard Hill',
+        addressLine4: 'The Royal Gala Valey'
+      }
     }
   }
-}];
+];
 
 experiment('internal/modules/billing/services/transactions-csv', async () => {
   experiment('_getTransactionData', async () => {
-    const transaction = invoicesForBatch[0].invoiceLicences[0].transactions[1];
+    const transaction = invoicesForBatch[0].invoiceLicences[0].transactions[0];
     test('returns transaction data in expected order', async () => {
       const transactionData = transactionsCSV._getTransactionData(transaction);
-      expect(transactionData[0]).to.equal(transaction.value);
-      expect(transactionData[1]).to.equal(transaction.isCredit.toString());
-      expect(transactionData[2]).to.equal(transaction.isCompensationCharge.toString());
-      expect(transactionData[3]).to.equal(transaction.chargeElement.source);
-      expect(transactionData[4]).to.equal(transaction.chargeElement.season);
-      expect(transactionData[5]).to.equal(transaction.chargeElement.loss);
-      expect(transactionData[6]).to.equal(transaction.description);
-      expect(transactionData[7]).to.equal(transaction.agreements[0]);
-      expect(transactionData[8]).to.equal(transaction.chargePeriod.startDate);
-      expect(transactionData[9]).to.equal(transaction.chargePeriod.endDate);
-      expect(transactionData[10]).to.equal(transaction.authorisedDays);
-      expect(transactionData[11]).to.equal(transaction.billableDays);
-      expect(transactionData[12]).to.equal('1 Apr');
-      expect(transactionData[13]).to.equal('31 Oct');
-      expect(transactionData[14]).to.equal(transaction.chargeElement.authorisedAnnualQuantity);
-      expect(transactionData[15]).to.equal(transaction.chargeElement.billableAnnualQuantity);
+      expect(transactionData.value).to.equal(transaction.value);
+      expect(transactionData.isCredit).to.equal(transaction.isCredit.toString());
+      expect(transactionData.isCompensationCharge).to.equal(transaction.isCompensationCharge.toString());
+      expect(transactionData.source).to.equal(transaction.chargeElement.source);
+      expect(transactionData.season).to.equal(transaction.chargeElement.season);
+      expect(transactionData.loss).to.equal(transaction.chargeElement.loss);
+      expect(transactionData.description).to.equal(transaction.description);
+      expect(transactionData.agreements[0]).to.equal(transaction.agreements[0]);
```

Comment:
Should this just be `transactionData.agreements`? Looking at the following test, this looks like you're only comparing the first character in the string here.

Reply:
Well spotted - thanks very much. I thought it returned an array but I guess it would help if I properly looked at what your code does. :-)